core/application/impl/app_config_impl.cpp

```diff
@@ -0,0 +1,332 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "application/impl/app_config_impl.hpp"
+
+#include <rapidjson/document.h>
+#include <rapidjson/filereadstream.h>
+#include <boost/program_options.hpp>
+#include <iostream>
+
+namespace {
+  struct RegisteredHandlers {
+    using Handler =
+        void (kagome::application::AppConfigurationImpl::*)(rapidjson::Value &);
+    char const *segment_name;
+    Handler handler;
+  };
+
+  template <typename T, typename Func>
+  inline void find_argument(boost::program_options::variables_map &vm,
+                            char const *name,
+                            Func &&f) {
+    assert(nullptr != name);
+    auto it = vm.find(name);
+    if (it != vm.end()) std::forward<Func>(f)(it->second.as<T>());
+  }
+
+  const std::string def_rpc_http_host = "0.0.0.0";
+  const std::string def_rpc_ws_host = "0.0.0.0";
+  const uint16_t def_rpc_http_port = 40363;
+  const uint16_t def_rpc_ws_port = 40364;
+  const uint16_t def_p2p_port = 30363;
+  const int def_verbosity = 2;
+  const bool def_is_only_finalizing = false;
+}  // namespace
+
+namespace kagome::application {
+
+  void AppConfigurationImpl::file_deleter(FILE *f) {
+    std::fclose(f);
+  }
+
+  AppConfigurationImpl::AppConfigurationImpl(kagome::common::Logger logger)
+      : logger_(std::move(logger)),
+        rpc_http_host_(def_rpc_http_host),
+        rpc_ws_host_(def_rpc_ws_host),
+        rpc_http_port_(def_rpc_http_port),
+        rpc_ws_port_(def_rpc_ws_port),
+        p2p_port_(def_p2p_port),
+        verbosity_(static_cast<spdlog::level::level_enum>(def_verbosity)),
+        is_only_finalizing_(def_is_only_finalizing) {}
+
+  AppConfigurationImpl::FilePtr AppConfigurationImpl::open_file(
+      const std::string &filepath) {
+    assert(!filepath.empty());
+    return AppConfigurationImpl::FilePtr(std::fopen(filepath.c_str(), "r"),
+                                         &file_deleter);
+  }
+
+  [[maybe_unused]] size_t AppConfigurationImpl::get_file_size(
+      const std::string &filepath) {
+    struct stat statbuf;
+    if (-1 == stat(filepath.c_str(), &statbuf)) {
+      logger_->error("Read configuration file metadata failed: {}", filepath);
+      exit(EXIT_FAILURE);
+    }
+    return static_cast<size_t>(statbuf.st_size);
+  }
+
+  bool AppConfigurationImpl::load_str(const rapidjson::Value &val,
+                                      char const *name,
```

Comment:
Pls, try to avoid raw pointer everywhere. E.g., we can use `std::string_view` and his method `.data()` inplace.

Reply:
I don't see the necessity of it in this place. Because we only forward ```name``` to ```rapidjson::Value::FindMember``` where ```StringRef``` created and it nullptr-safe. 
More over ```string_view``` ctor initializes length-member, than we have string scanning for null-termination. I can add compile-time length calculation for literal const values to avoid string-scan, but it will increase code complexity without any profit.